staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1beta1/types.go

```diff
@@ -53,13 +53,23 @@ type APIServiceSpec struct {
 	// CABundle is a PEM encoded CA bundle which will be used to validate an API server's serving certificate.
 	CABundle []byte `json:"caBundle" protobuf:"bytes,5,opt,name=caBundle"`
 
-	// Priority controls the ordering of this API group in the overall discovery document that gets served.
-	// Client tools like `kubectl` use this ordering to derive preference, so this ordering mechanism is important.
-	// Values must be between 1 and 1000
-	// The primary sort is based on priority, ordered lowest number to highest (10 before 20).
+	// GroupPriority controls the ordering of the group with respect to other groups. The lowest of all GroupPriorities
+	// for the same group is used.  Must be greater than zero.
+	// The primary sort is based on GroupPriority, ordered lowest number to highest (10 before 20).
+	// The secondary sort is based on the VersionPriority, ordered lowest number to highest.
+	// The tertiary sort is based on the alphabetical comparison of the name of the object.  (v1.bar before v1.foo)
+	// We'd recommend something like: *.k8s.io (except extensions) at 1000, extensions at 2000
+	// PaaSes (OpenShift, Deis) are recommended to be in the 3000s
+	GroupPriority int32 `json:"groupPriority" protobuf:"varint,7,opt,name=groupPriority"`
+
+	// VersionPriority controls the ordering of this API version inside of its group.
+	// The primary sort is based on VersionPriority, ordered lowest number to highest (10 before 20).
 	// The secondary sort is based on the alphabetical comparison of the name of the object.  (v1.bar before v1.foo)
-	// We'd recommend something like: *.k8s.io (except extensions) at 100, extensions at 150
-	// PaaSes (OpenShift, Deis) are recommended to be in the 200s
+	// Since it's inside of a group, the number can be small, probably in the 10s
+	VersionPriority int32 `json:"versionPriority" protobuf:"varint,8,opt,name=versionPriority"`
+
+	// Priority is completely ignored.  It never appeared in any permanent level of kubernetes, but it was in a beta API in a beta
+	// tagged level, so we tolerate the field being present.
 	Priority int64 `json:"priority" protobuf:"varint,6,opt,name=priority"`
```

Comment:
removing doesn't cause problems with anything other than client-side validation, right? any other reason to keep it?